processing/src/main/java/org/apache/druid/segment/SimpleQueryableIndex.java

```diff
@@ -42,19 +44,20 @@
   private final List<String> columnNames;
   private final Indexed<String> availableDimensions;
   private final BitmapFactory bitmapFactory;
-  private final Map<String, ColumnHolder> columns;
+  private final Map<String, Supplier<ColumnHolder>> columns;
   private final SmooshedFileMapper fileMapper;
   @Nullable
   private final Metadata metadata;
-  private final Map<String, DimensionHandler> dimensionHandlers;
+  private final Supplier<Map<String, DimensionHandler>> dimensionHandlers;
 
   public SimpleQueryableIndex(
       Interval dataInterval,
       Indexed<String> dimNames,
       BitmapFactory bitmapFactory,
-      Map<String, ColumnHolder> columns,
+      Map<String, Supplier<ColumnHolder>> columns,
       SmooshedFileMapper fileMapper,
-      @Nullable Metadata metadata
+      @Nullable Metadata metadata,
+      boolean lazy
```

Comment:
It would be preferable lazy handling be done in one place, i.e. IndexIO . So, maybe take dimensionHandlers as arg ?

```suggestion
      Supplier<Map<String, DimensionHandler>> dimensionHandlers
```

Reply:
I think keep current is acceptable